validation-test/StdlibUnittest/ChildProcessShutdown/RequireOwnProcess.swift

```diff
@@ -0,0 +1,35 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+
+import StdlibUnittest
+#if os(macOS) || os(iOS) || os(tvOS) || os(watchOS)
+  import Darwin
+#elseif os(Linux) || os(FreeBSD) || os(PS4) || os(Android) || os(Cygwin) || os(Haiku)
+  import Glibc
+#elseif os(Windows)
+  import MSVCRT
+#else
+#error("Unsupported platform")
+#endif
+
+//
+// Test that a test runs in its own child process if asked.
+//
+
+enum Globals {
+  static var childProcessWasExecuted = false
+}
+
+var TestSuiteRequireNewProcess = TestSuite("TestSuiteRequireNewProcess")
+
+TestSuiteRequireNewProcess.test("RequireOwnProcess")
+  .requireOwnProcess()
+  .code {
+  Globals.childProcessWasExecuted = true
+}
+
+TestSuiteRequireNewProcess.test("ShouldNotReusePreviousProcess") {
+  expectEqual(false, Globals.childProcessWasExecuted)
```

Comment:
Nitpick: You should test this in both directions by having a test *before* `RequireOwnProcess` too.